tools/adventure-pack/goodies/java/UnionFind.java

```diff
@@ -0,0 +1,24 @@
+class UnionFind {
+    
+    private int[] parent;
+    
+    public UnionFind(int n){
+        parent = new int[n];
+        for(int i = 0; i < n; i++){
+            parent[i] = i;
+        }
+    }
+    
+    public void union(int a, int b){
+        int pa = find(a);
+        int pb = find(b);
+        
+        parent[pa] = parent[pb];
```

Comment:
Thanks for the PR! As a follow-up (doesn't have to be in this PR) it would be nice to add the behavior that the larger group is the one that absorbs the smaller group during a union operation.

Reply:
Ack!